code/modules/nightmare/nmtasks/mapscheduler.dm

```diff
@@ -0,0 +1,35 @@
+/// Handles map insertions sequentially and updating the game to match map insertions
+/datum/nmtask/scheduler/mapload
+	name = "mapload scheduler"
+	/// Map boundaries tainted by children nightmare tasks to be handled
+	var/list/tainted_bounds = list()
+
+/datum/nmtask/scheduler/mapload/execute()
+	. = ..()
+	makepowernets()
+	repopulate_sorted_areas()
+	patch_lighting()
+
+/datum/nmtask/scheduler/mapload/add_task(datum/nmtask/task)
+	. = ..()
+	RegisterSignal(task, COMSIG_NIGHTMARE_TAINTED_BOUNDS, .proc/register_tainted_bounds)
+
+/datum/nmtask/scheduler/mapload/proc/register_tainted_bounds(datum/nmtask/task, list/bounds)
+	tainted_bounds.len++
+	tainted_bounds[tainted_bounds.len] = bounds
+
+/datum/nmtask/scheduler/mapload/proc/patch_lighting()
+	var/list/tainted = list()
+
+	for(var/list/bounds as anything in tainted_bounds)
+		var/list/TT = 	block(	locate(bounds[MAP_MINX], bounds[MAP_MINY], bounds[MAP_MINZ]),
```

Comment:
```suggestion
		var/list/TT = block(locate(bounds[MAP_MINX], bounds[MAP_MINY], bounds[MAP_MINZ]),
```
is this spacing important? i know block is a fucky little define, so lmk

Reply:
oh no that's probably an accidental leftover from refactoring similar code in another version